### PR TITLE
[Improvement] Overhaul Stats and Augmentations UI

### PR DESCRIPTION
**Pull request by nickofolas**
<span id="top">tl;dr overhauls stats and augmentations page UI, as per title</span>

this PR completely redesigns the UI for these 2 pages with functionality and visual appeal in mind

> **mini table of contents (a lot of screenshots to go through)**
> 1. [Stats UI](#user-content-stats-ui)
> 2. [Augmentations Page UI](#user-content-augmentations-ui)
> 3. [Dev Menu Improvements](#user-content-dev-menu)
> 4. [Testing](#user-content-testing)

<h1 id="stats-ui">Stats UI</h2>

[To Top](#user-content-top)

the stats page has been updated to be more compact, while also presenting information in a way that should be easy to digest.

image
*the new stats page as seen from a brand new save*

information is now arranged in a way that allows as more stats to be displayed without having to scroll the page. in addition, stats have been colorized in accordance with UI elements such as the character overview.

the new stats UI is also adaptive as the player progresses through the game.

image
*once the player has destroyed a bitnode, information about the current bitnode (and level) is added to the bottom of the page*

image
image
*job stats have been moved to a popup-based system*

image
*bladeburner stats are shown when available*

image
image
*when Source-File 5 is owned, BitNode multipliers will be shown as well*

image
*intelligence is displayed when it is enabled*

<h1 id="augmentations-ui">Augmentations UI</h1>

[To Top](#user-content-top)

the Augmentations page UI has also been redesigned:
* the augs page no longer shows all stats, since the stats page already does this
* neuroflux governor and entropy levels have been moved to a separate section, as they are too unique to group with other augmentations

image
*the new augmentations UI from a brand new save*

image
*the new BitNode display*

image
*the new entropy virus display*

image
*the new neuroflux governor display*

image
*neuroflux governor **together with** entropy*

image
image
image
image
*the new Purchased Augmentations display*

**Note:** the new Purchased Augmentations display will __only__ show stats that have been changed.

image
*the new Installed Augmentations display*

Source Files and Augmentations will continue to **both** be sorted by the sorting buttons for Installed Augmentations.

firefox_iaPVhplvGL

a fully-utilized Aug page example

<h1 id="dev-menu">Dev Menu Improvement</h1>

[To Top](#user-content-top)

Added a button to the Augmentations section of the dev menu to instantly clear all queued Augmentations

image

<h1 id="testing">Testing</h1>

[To Top](#user-content-top)

as shown by the screenshots, everything appears to be working

beyond simply looking at the UI, i've tested as follows:
- [x] NFG/entropy display works with any combination of the two
- [x] purchased aug stats change window is properly dynamic with the layout of the stats
- [x] information does not appear when it shouldn't (bitnode mults, bitnode info, etc)
- [x] Source-File -1 works properly
- [x] i've tested and confirmed that functionality is retained, and as far as i can tell, that is indeed the case

in the event that i've missed glaring issues, i'll make sure to fix any reported immediately

### CODE REVIEW

**Comment by phyzical** [comment]
looking snazzy as always, ill review this later today

File: src/DevMenu/ui/Augmentations.tsx [hunk 1]
```diff
@@ -39,50 +38,46 @@ export function Augmentations(props: IProps): React.ReactElement {
...
+          }
+        >
+          {Object.values(AugmentationNames).map((aug) => (
+            <MenuItem key={aug} value={aug}>
+              {aug}
+            </MenuItem>
+          ))}
+        </Select>
+        <Button sx={{ display: "block" }} onClick={clearQueuedAugs}>
+          Clear Queued Augmentations
```
Comment (phyzical) [inline src/DevMenu/ui/Augmentations.tsx:79]: should we add note explaining you dont get refunded?

Comment (nickofolas) [reply src/DevMenu/ui/Augmentations.tsx:79]: I think it's fine. They're already in the dev menu, it's not like they're losing anything :p

Comment (phyzical) [reply src/DevMenu/ui/Augmentations.tsx:79]: oh derp good point 😆

**Pull request merged**